### PR TITLE
feat: migrate to material3

### PR DESCRIPTION
**Pull request by aizensoosuke**
#37

### CODE REVIEW

File: app/src/main/java/com/github/studydistractor/sdp/login/FirebaseLoginAuth.kt [hunk 1]
```diff
@@ -11,6 +12,7 @@ import javax.inject.Inject
 class FirebaseLoginAuth @Inject constructor() : LoginAuthInterface {
     lateinit var auth : FirebaseAuth
     override fun loginWithEmail(email: String, password: String): Task<AuthResult> {
+        auth = FirebaseAuth.getInstance();
```
Comment (bluedrack) [inline app/src/main/java/com/github/studydistractor/sdp/login/FirebaseLoginAuth.kt:15]: I think this change is not necessary.

**Review by bluedrack — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/github/studydistractor/sdp/login/FirebaseLoginAuth.kt (hunk 1 above)
Comment (aizensoosuke) [reply app/src/main/java/com/github/studydistractor/sdp/login/FirebaseLoginAuth.kt:15]: I'm getting errors that auth is null

Comment (lipefree) [reply app/src/main/java/com/github/studydistractor/sdp/login/FirebaseLoginAuth.kt:15]: How do you handle the init of auth otherwise @bluedrack ?

Comment (bluedrack) [reply app/src/main/java/com/github/studydistractor/sdp/login/FirebaseLoginAuth.kt:15]: No, it is the good fix but I tought it was not the right commit to fix it. Nevermind

**Review by AthenaPK — DISMISSED** [review]
LGTM!

**Comment by aizensoosuke** [comment]
This has been resolved in the other MRs of this sprint.

**Pull request closed without merging**